src/test/java/com/microsoft/azure/relay/SendReceiveTest.java

```diff
@@ -106,19 +103,19 @@ public void websocketRepeatedSendReceiveTest() {
 		CompletableFuture<Integer> senderReceiveCount = new CompletableFuture<Integer>();
 		CompletableFuture<Integer> listenerReceiveCount = new CompletableFuture<Integer>();
 		
-		listener.acceptConnectionAsync().thenAcceptAsync((websocket) -> {
+		listener.acceptConnectionAsync().thenAcceptAsync(websocket -> {
 			for (int i = 1; i <= timesToRepeat; i++) {
 				websocket.readAsync().join();
-				websocket.writeAsync(StringUtil.toBuffer("hi")).join();
+				websocket.writeAsync(ByteBuffer.wrap(SMALL_BYTES)).join();
```

Comment:
>ByteBuffer.wrap(SMALL_BYTES)). [](start = 25, length = 30)

What impact does this comment from ClientWebSocket.java have on the shared SMALL_BYTES array?
"sendBinary() will cause the content of the byte array within the ByteBuffer to change"

Reply:
is that why some places call .clone()?

---
In reply to: [295925633](https://github.com/Azure/azure-relay-java/pull/47#discussion_r295925633) [](ancestors = 295925633)

Reply:
Yes, `.clone()` just to avoid the content getting messed up

---
In reply to: [295927864](https://github.com/Azure/azure-relay-java/pull/47#discussion_r295927864) [](ancestors = 295927864,295925633)

Reply:
Let's put clone into our ClientWebSocket then.

---
In reply to: [295930809](https://github.com/Azure/azure-relay-java/pull/47#discussion_r295930809) [](ancestors = 295930809,295927864,295925633)